### PR TITLE
Support Class.forName(Module, String) in abstract interpretation

### PR DESCRIPTION
**Pull request by johannesduesing**
**Reason for this PR**
This PR is to support tracking of Class values in abstract interpretation when `Class.forName(Module, String)` or `Class.forName(Module, String, Class)` are invoked, see #87.

**Changes in this PR**
`org.opalj.ai.domain.l1.ClassValues` now explicitly handles calls to the new `Class.forName` methods involving Modules introduced in Java 9.  The actual behavior of these methods is as follows: If the class with the given name is part of the given Module, then the class is returned, otherwise `null` is returned. For the abstract interpretation we overapproxmiate this behavior and always return the class with the given name. This is because currently there is no tracking of values of type `Module`, which would allow us to actually check whether the class belongs to the given module or not.

**Note**
Testing this implementation requires me to compile the test fixtures `ai.jar` with Java 9 instead of Java 8 (because `Class.forName(Module, String)` was introduced in Java 9). This lead to the fixture jar being renamed to `ai-9.jar` (i did not find a fixture compiler option to specify the output filename) which in turn caused a lot of test files being changed, since i had to replace `"ai.jar"` with `"ai-9.jar"`.

### CODE REVIEW

File: OPAL/ai/src/main/scala/org/opalj/ai/domain/l1/ClassValues.scala [hunk 1]
```diff
@@ -184,8 +189,23 @@ trait ClassValues
                             )
                     }
 
+                case _ if secondOperandIsStringValue =>
```
Comment (errt) [inline OPAL/ai/src/main/scala/org/opalj/ai/domain/l1/ClassValues.scala]: I wonder if this matching could be done in a nicer way, maybe along the lines of
```
operands match {
  case _ :+ (sv: StringValue) =>
  case _ :+ (sv: StringValue) :+ _ =>
}
```
?

Comment (errt) [reply OPAL/ai/src/main/scala/org/opalj/ai/domain/l1/ClassValues.scala]: Or maybe do away with the value match and just match on the descriptor?

**Review by errt — DISMISSED** [review]

**Comment by errt** [comment]
Renamed tests are okay, another option would have been to separate out the new tests into their own fixture.

File: OPAL/ai/src/main/scala/org/opalj/ai/domain/l1/ClassValues.scala (hunk 1 above)
Comment (johannesduesing) [reply OPAL/ai/src/main/scala/org/opalj/ai/domain/l1/ClassValues.scala]: I now changed this so it uses `operands.reverse match { ... }` - should essentially be the same.

Comment (errt) [reply OPAL/ai/src/main/scala/org/opalj/ai/domain/l1/ClassValues.scala]: It creates a reversed list, though. Probably not a performance issue here, though.

Comment (johannesduesing) [reply OPAL/ai/src/main/scala/org/opalj/ai/domain/l1/ClassValues.scala]: Ah yes that's right - i changed that to your suggestion. Thought the reversed version would be slightly more readable, as the first operand would actually be the first in the match clauses, but i think performance should be more important here. Could you re-approve your review so i can merge?

Comment (errt) [reply OPAL/ai/src/main/scala/org/opalj/ai/domain/l1/ClassValues.scala]: Did you make sure it works? Readability can be enhanced by commenting on what the cases do.

**Review by errt — APPROVED** [review]

File: OPAL/ai/src/main/scala/org/opalj/ai/domain/l1/ClassValues.scala (hunk 1 above)
Comment (johannesduesing) [reply OPAL/ai/src/main/scala/org/opalj/ai/domain/l1/ClassValues.scala]: Well i ran all the new unit tests and it does yield the desired results - it seems to be functionally equivalent to the previous solutions. Comments for readability are already in place, so i think we're fine there.

**Pull request merged**